libsolidity/ast/AST.h

```diff
@@ -684,12 +688,15 @@ class UsingForDirective: public ASTNode
 
 	/// @returns a list of functions or the single library.
 	std::vector<ASTPointer<IdentifierPath>> const& functionsOrLibrary() const { return m_functions; }
+	auto functionsAndOperators() const { return ranges::zip_view(m_functions, m_operators); }
 	bool usesBraces() const { return m_usesBraces; }
 	bool global() const { return m_global; }
 
 private:
 	/// Either the single library or a list of functions.
 	std::vector<ASTPointer<IdentifierPath>> m_functions;
+	/// Operators, the functions are applied to.
+	std::vector<std::optional<Token>> m_operators;
```

Comment:
ok. good point about them not being unique. I was thinking mostly about the case where you have multiple functions attached to the same operator and we could mostly disallow that - but not for `-` so yeah, it can't be a map after all.